doc/developer/code-coverage.md

```diff
@@ -0,0 +1,50 @@
+# Code Coverage for Pull Requests
+
+## Motivation
+
+A full [code coverage report](http://65.109.125.29/coverage_01879007-43c4-49db-a81d-a44e93dfb55a/) allows you to check for each part of the code if there is a test that currently covers it by executing the relevant lines of code. Unfortunately the motivation to increase code coverage using full reports is low, and the effort to understand why no existing mzcompose-based test covers the code in question is high.
+
+Thus to prevent regressions and new bugs we instead want to ensure that new and changed productive code in a pull request (PR) is adequately tested. This makes sense both when code is refactored as well as when new functionality is added. For each PR you or someone from the QA team can use the [Coverage](https://buildkite.com/materialize/coverage/builds) Buildkite pipeline to trigger a coverage build and test run with coverage information for it. After the test runs the resulting coverage information will be aggregated and analyzed to indicate which changed lines of productive code are not covered.
+
+## Triggering Coverage for your PR
+
+As an example let's take the *[adapter: Update privileges when an owner is updated](https://github.com/MaterializeInc/materialize/pull/18777)* pull request. To run PR code coverage against it, go to the [Coverage Buildkite pipeline](https://buildkite.com/materialize/coverage), click on "New Build" and enter the full commit hash and branch with your Github username:
+
+![Commit: 69de84971e8c16ab459c88be73870bb676dbc478, Branch: jkosh44:update-privilege-owners](assets/coverage-new-build.png)
+
+After some time your [Coverage run](https://buildkite.com/materialize/coverage/builds/26) will be finished and contain a summary of any uncovered lines in the form of a `git diff` where each removed line is not covered:
```

Comment:
I was thinking about this the other day. It may be easier to explain this if we used `git diff --output-indicator-old=!` , to use "!", a neutral yet alarming symbol, so that people are not confused by the "-". This will slightly reduce the cognitive load required to process the diff. (no need to do this in this PR)

Reply:
Great suggestion 👍, I was also thinking about replacing the `-` with something else (but was not aware that this can already be specified in the git command).

Reply:
Good idea, that confused people already, will do so in follow-up.